src/webui/api/appcontroller.cpp

```diff
@@ -576,9 +633,145 @@ void AppController::setPreferencesAction()
         RSS::Session::instance()->setProcessingEnabled(it.value().toBool());
     if (hasKey("rss_auto_downloading_enabled"))
         RSS::AutoDownloader::instance()->setProcessingEnabled(it.value().toBool());
+
+    // Advanced settings
+    // qBittorrent preferences
+    // Current network interface
+    if (hasKey("current_network_interface")) {
+        QString ifaceName {};
+        const QString ifaceValue {it.value().toString()};
+
+        for (const QNetworkInterface &iface : asConst(QNetworkInterface::allInterfaces()))
```

Comment:
seems you can use `std::find_if()`

Reply:
I give up ... The program always crashes at `iface.addressEntries()` when `ifaceValue` is empty (for "All interfaces").

```
const auto ifaceNameIter = std::find_if(
    std::begin(QNetworkInterface::allInterfaces()), std::end(QNetworkInterface::allInterfaces()),
    [&](const QNetworkInterface& iface) {
        return (!iface.addressEntries().isEmpty()) && (iface.name() == ifaceValue);
    }
);
```

Reply:
try this:
```c++
const QList<QNetworkInterface> ifaces = QNetworkInterface::allInterfaces();
const auto ifacesIter = std::find_if(ifaces.cbegin(), ifaces.cend(),
    [&ifaceValue](const QNetworkInterface &iface) {
        return (!iface.addressEntries().isEmpty()) && (iface.name() == ifaceValue);
    });
ifaceName = (ifacesIter != ifaces.cend()) ? ifacesIter->humanReadableName() : {};
```